### ISSUE TITLE
ImageItem float luts "regression"

### ISSUE THREAD

**Issue by pijyoi** [pyqtgraph/pyqtgraph#1984]
ImageItem weakly documents that the lut that it accepts is of type ubyte:
https://pyqtgraph.readthedocs.io/en/latest/graphicsItems/imageitem.html#pyqtgraph.ImageItem.setLookupTable
https://pyqtgraph.readthedocs.io/en/latest/functions.html#pyqtgraph.makeARGB

However, in pyqtgraph <= 0.12.1, float luts scaled to 255 are silently accepted and work.
Since pyqtgraph >= 0.12.2, such float luts no longer work in certain cases. (lut more than 256 entries, image data is floating)
luts that are ubyte work as always. 

Sample demonstrating code. (Matplotlib returns float luts)
```python
import pyqtgraph as pg
import numpy as np
import matplotlib.pyplot as plt

app = pg.mkQApp()
glw = pg.GraphicsLayoutWidget()
glw.show()

# affected types: float32, float64, uint16
ramp = np.arange(1000, dtype=np.float32)
ramp = ramp.reshape((1, -1))
cmap = plt.get_cmap("viridis")

for idx, num_colors in enumerate([256, 257]):
    lut = cmap(np.linspace(0, 1, num_colors)) * 255
    # lut = lut.astype(np.uint8)

    imgitem = pg.ImageItem(ramp, axisOrder='row-major')
    imgitem.setLookupTable(lut)

    vb = glw.addViewBox(row=idx, col=0)
    vb.addItem(imgitem)

app.exec_()
```
Output for pyqtgraph >= 0.12.2
For pyqtgraph <= 0.12.1, both bars are smooth gradients.
![Capture](https://user-images.githubusercontent.com/2657027/132125347-58bd01dc-c9ca-4763-9bf5-473f17dad438.PNG)

Possible "fixes":
1) Do nothing
2) Fail noisily if lut is not uint8
3) Warn if lut is not uint8
4) Cast the float ndarray to uint8 within the library

Note that even though float luts work now if the number of colors is <= 256, the library is forced to go through the slower ```makeARGB()``` codepath.

**Comment by j9ac9k** [comment]
@outofculture  what do you think?

**Comment by pijyoi** [comment]
This issue was motivated when I chanced upon https://github.com/acconeer/acconeer-python-exploration/issues/88
After experimenting some more, I think I have managed to reproduce the error encountered in the referenced issue.
i.e. an exception is thrown rather than the wrong colorbar being displayed.

The triggering conditions are:
1) row-major floating image data type
2) col-major ImageItem
   * note that this is a pessimization
3) scaled floating lut with > 256 entries
   * note that lut is supposed to be uint8
   * note that the lut had > 256 entries because matplotlib viridis comes with 259 entries and the code had obtained the lut by referencing matplotlib private variables (https://github.com/acconeer/acconeer-python-exploration/blob/15df50d57ecc197501fbf9c5838bc0293bc3d8e2/src/acconeer/exptool/utils.py#L395) 

```python
import pyqtgraph as pg
import numpy as np
import matplotlib.pyplot as plt

app = pg.mkQApp()
glw = pg.GraphicsLayoutWidget()
glw.show()

# affected types: float32, float64, uint16
ramp = np.arange(1000, dtype=np.float32)
data = np.column_stack((ramp, ramp))
cmap = plt.get_cmap("viridis")

for idx, num_colors in enumerate([256, 257]):
    lut = cmap(np.linspace(0, 1, num_colors)) * 255
    # lut = lut.astype(np.uint8)

    imgitem = pg.ImageItem(data, axisOrder='col-major')
    imgitem.setLookupTable(lut)

    vb = glw.addViewBox(row=idx, col=0)
    vb.addItem(imgitem)

app.exec_()
```

**Comment by pijyoi** [comment]
"fixed" by #1985
- deprecation warning emitted for non-uint8 LUTs
- rendering for non-uint8 LUTs handed to makeARGB()
